internal/services/contracts_events_consumer_test.go

```diff
@@ -0,0 +1,264 @@
+	defer s.destroy()
+
+	e := eventsPayloadFactory(2, 2, "SomeEvent", 0)
+	factoryResp := e[0]
+
+	msg := &message.Message{
+		Payload: []byte(factoryResp.payload),
+	}
+	c := NewContractsEventsConsumer(s.pdb, &s.logger, s.settings)
+
+	err := c.processMessage(msg)
+	s.assert.NoError(err)
+
+	s.assert.Nil(err)
+
+	args := factoryResp.args
+
+	nft, err := models.FindNFTPrivilege(s.ctx, s.pdb.DBS().Reader, args.contract.Bytes(), args.tokenID, args.privilegeID, args.userAddress.Bytes())
+	s.assert.EqualError(err, "sql: no rows in result set")
+
+	s.assert.Nil(nft)
+}
+
+func TestUpdatedTimestamp(t *testing.T) {
+	s := initCEventsTestHelper(t)
+	defer s.destroy()
+
+	e := eventsPayloadFactory(3, 3, "", 0)
+	factoryResp := e[0]
+
+	c := NewContractsEventsConsumer(s.pdb, &s.logger, s.settings)
+
+	msg := &message.Message{
+		Payload: []byte(factoryResp.payload),
+	}
+	err := c.processMessage(msg)
+	s.assert.NoError(err)
+
+	args := factoryResp.args
+
+	oldNft, err := models.FindNFTPrivilege(s.ctx, s.pdb.DBS().Reader, args.contract.Bytes(), args.tokenID, args.privilegeID, args.userAddress.Bytes())
+	s.assert.NoError(err)
+
+	s.assert.NotNil(oldNft)
+
+	expiry := time.Now().Add(time.Hour + time.Duration(4)).UTC().Unix()
+	e = eventsPayloadFactory(3, 3, "", expiry)
+	factoryResp = e[0]
+
+	msg = &message.Message{
+		Payload: []byte(factoryResp.payload),
+	}
+	err = c.processMessage(msg)
+	s.assert.NoError(err)
+
+	a, _ := models.NFTPrivileges().All(s.ctx, s.pdb.DBS().Reader)
+	s.assert.Equal(len(a), 1)
+
+	newNft, err := models.FindNFTPrivilege(s.ctx, s.pdb.DBS().Reader, args.contract.Bytes(), args.tokenID, args.privilegeID, args.userAddress.Bytes())
+	s.assert.NoError(err)
+
+	actual := mockTestEntity{
+		Contract:    newNft.ContractAddress,
+		TokenID:     newNft.TokenID,
+		PrivilegeID: newNft.Privilege,
+		UserAddress: newNft.UserAddress,
+		ExpiresAt:   newNft.Expiry,
+	}
+
+	expected := mockTestEntity{
+		Contract:    args.contract.Bytes(),
+		UserAddress: args.userAddress.Bytes(),
+		TokenID:     args.tokenID,
+		ExpiresAt:   time.Unix(expiry, 0).UTC(),
+		PrivilegeID: args.privilegeID,
+	}
+
+	s.assert.Equal(expected, actual, "Event was updated successful")
+	s.assert.Equal(oldNft.CreatedAt, newNft.CreatedAt)
+	s.assert.NotEqual(oldNft.UpdatedAt, newNft.UpdatedAt)
+}
+
+// Utility/Helper functions
+func eventsPayloadFactory(from, to int, eventName string, exp int64) []eventsFactoryResp {
+	res := []eventsFactoryResp{}
+
+	convertTokenIDToDecimal := func(t string) types.Decimal {
+		ti, ok := new(decimal.Big).SetString(t)
+		if !ok {
+			return types.Decimal{}
+		}
+
+		return types.NewDecimal(ti)
+	}
+
+	if eventName == "" {
+		eventName = "PrivilegeSet"
+	}
+
+	for i := from; i <= to; i++ {
+		contractAddr := common.BytesToAddress([]byte{uint8(i)})
+		userAddr := common.BytesToAddress([]byte{uint8(i + 1)})
+		tokenID := convertTokenIDToDecimal(fmt.Sprint(i))
+		privID := i + 1
+		expiry := time.Now().Add(time.Hour + time.Duration(i)).UTC().Unix()
```

Comment:
Changing this to
```suggestion
		expiry := time.Now().Add(time.Hour + time.Duration(i)*time.Minute).UTC().Unix()
```
makes the whitelist matter.